### PR TITLE
Add enablement tests for feature gate nodeInclusionPolicyInPodTopologySpread

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind feature
/sig scheduling

#### What this PR does / why we need it:

Add feature gate enablement tests for nodeInclusionPolicy.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- KEP: https://github.com/kubernetes/enhancements/pull/3539
```

### CODE REVIEW

File: pkg/registry/core/pod/strategy_test.go [hunk 1]
```diff
@@ -1435,3 +1440,89 @@ func TestDropNonEphemeralContainerUpdates(t *testing.T) {
...
+			}
+			wantPod := pod.DeepCopy()
+			pod.Spec.TopologySpreadConstraints = append(pod.Spec.TopologySpreadConstraints, tc.topologySpreadConstraints...)
+
+			errs := Strategy.Validate(genericapirequest.NewContext(), pod)
+			if len(errs) != 0 {
+				t.Errorf("Unexpected error: %v", errs.ToAggregate())
+			}
+
+			Strategy.PrepareForCreate(genericapirequest.NewContext(), pod)
```
Comment (kerthcet) [inline pkg/registry/core/pod/strategy_test.go:1521]: We'll call `DropDisabledPodFields` in `PrepareForCreate` which will processes below logic and helps us to verify the feature gate enablement. https://github.com/kubernetes/kubernetes/blob/ecdbb0258049b9c902b53c77d386b741e2a25cb7/pkg/api/pod/util.go#L596-L609

Comment (wojtek-t) [reply pkg/registry/core/pod/strategy_test.go:1521]: This is good test, but it's still not exactly what I wanted.
The crucial part of https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282 that I linked you is the fact that the feature gate is changed in the middle of the test, so basically:
- we store an object with a new field
- we disabled the FG
- we check if updating the object with the FG now disabled works as expected now..

**Comment by kerthcet** [comment]
@wojtek-t just added another test for updating, plz take a look.

File: pkg/registry/core/pod/strategy_test.go (hunk 1 above)
Comment (kerthcet) [reply pkg/registry/core/pod/strategy_test.go:1521]: Kindly ping @wojtek-t , added a new test as advised.

**Comment by kerthcet** [comment]
Kindly ping @wojtek-t

**Comment by wojtek-t** [comment]
Thanks for adding these tests.

/lgtm
/approve

**Comment by kerthcet** [comment]
The unit test is not related, but I'll take a look. Trace the log here https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/112805/pull-kubernetes-unit/1588453554656382976

/retest

**Comment by kerthcet** [comment]
It's a known issue here https://github.com/kubernetes/kubernetes/issues/107414

**Pull request merged**